routers/matchmaking.js

```diff
@@ -231,21 +232,30 @@ async function SetInstances(RoomId, InstanceList) {
 async function SetInstance(RoomId, InstanceId, Instance) {
     if(!Object.keys(SubroomInstances).includes(RoomId)) SubroomInstances[RoomId] = [];
 
-    var index = SubroomInstances[RoomId].findIndex(item => item.InstanceId === InstanceId);
+    var index = SubroomInstances[RoomId].findIndex(item => item.InstanceId == InstanceId);
     if(index < 0) SubroomInstances[RoomId].push(Instance);
     else SubroomInstances[RoomId][index] = Instance;
 }
 
 async function CreateInstance(RoomId, SubroomId, MatchmakingMode, TTL, Persistent, MaxPlayers) {
     if(!Object.keys(SubroomInstances).includes(RoomId)) SubroomInstances[RoomId] = [];
 
-    const room = new RoomSession(RoomId, SubroomId, MatchmakingMode, TTL, true, MaxPlayers);
+    const room = new RoomSession(RoomId, SubroomId, MatchmakingMode, TTL, Persistent, MaxPlayers);
```

Comment:
Fixed an issue where instances would always be persistent, this was a temporary fix to another problem where instances would close too early, which turned out to be a logic error, however this stopgap measure was never removed.